packages/core/src/components/fields/LayoutGridField.tsx

```diff
@@ -140,6 +139,594 @@ function isNumericIndex(str: string) {
+    <GridTemplate
+      column
+      key={`column-${hashObject(child)}`}
+      data-testid={LAYOUT_GRID_FIELD_TEST_IDS.col}
+      {...gridProps}
+    >
+      <LayoutGridFieldChildren {...layoutGridFieldProps} childrenLayoutGridSchemaId={[child]} />
+    </GridTemplate>
+  ));
+}
+
+/** Renders a `GridTemplate` as a container. The `layoutGridSchema` for the `GridType.ROW` is separated into the
+ * `children` and other `gridProps`. The `gridProps` will be spread onto the outer `GridTemplate`. Inside of the
+ * `GridTemplate` all of the `children` are rendered.
+ *
+ * @returns - The rendered `GridTemplate` containing the children for the `GridType.ROW`
+ */
+function LayoutGridRow<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
+  props: LayoutFieldProps<T, S, F>,
+) {
+  const { layoutGridSchema, ...layoutGridFieldProps } = props;
+
+  const { registry, uiSchema } = layoutGridFieldProps;
+  const { children, gridProps } = findChildrenAndProps<T, S, F>(layoutGridSchema, GridType.ROW, registry);
+  const uiOptions = getUiOptions<T, S, F>(uiSchema);
+  const GridTemplate = getTemplate<'GridTemplate', T, S, F>('GridTemplate', registry, uiOptions);
+
+  return (
+    <GridTemplate {...gridProps} data-testid={LAYOUT_GRID_FIELD_TEST_IDS.row}>
+      <LayoutGridFieldChildren {...layoutGridFieldProps} childrenLayoutGridSchemaId={children} />
+    </GridTemplate>
+  );
+}
+
+/**
+ * The props for the LayoutGridFieldChildren component.
+ */
+type LayoutGridFieldChildrenProps<
+  T = any,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any,
+> = LayoutGridFieldProps<T, S, F> & {
+  /** The list of strings or objects that represents the configurations for the children fields */
+  childrenLayoutGridSchemaId: LayoutGridSchemaType[];
+};
+
+/** Iterates through all the `childrenLayoutGridSchemaId`, rendering a nested `LayoutGridField` for each item in the
+ * list, passing all the props for the current `LayoutGridField` along, updating the `schema` by calling
+ * `retrieveSchema()` on it to resolve any `$ref`s. In addition to the updated `schema`, each item in
+ * `childrenLayoutGridSchemaId` is passed as `layoutGridSchema`.
+ *
+ * @returns - The nested `LayoutGridField`s
+ */
+function LayoutGridFieldChildren<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
+  props: LayoutGridFieldChildrenProps<T, S, F>,
+) {
+  const { childrenLayoutGridSchemaId, ...layoutGridFieldProps } = props;
+  const { registry, schema: rawSchema, formData } = layoutGridFieldProps;
+  const { schemaUtils } = registry;
+  const schema = schemaUtils.retrieveSchema(rawSchema, formData);
+  return childrenLayoutGridSchemaId.map((layoutGridSchema) => (
+    <LayoutGridField<T, S, F>
+      {...layoutGridFieldProps}
+      key={`layoutGrid-${hashObject(layoutGridSchema)}`}
+      schema={schema}
+      layoutGridSchema={layoutGridSchema}
+    />
+  ));
+}
```

Comment:
You could move this one above all the other components that use it. That way there this one component is the one that references `LayoutGridField` before it is used. All the others that use it won't have that error